### PR TITLE
Add Iran Timezone test case

### PR DESCRIPTION
**Pull request by tarekgh**

### CODE REVIEW

**Comment by tarekgh** [comment]
#17393

**Comment by tarekgh** [comment]
CC @eerhardt

File: src/System.Runtime/tests/System/TimeZoneInfoTests.cs [hunk 1]
```diff
@@ -311,6 +312,15 @@ public static void ConvertTime_DateTimeOffset_VariousSystemTimeZones()
             time3 = new DateTime(2003, 10, 26, 3, 0, 1);
             VerifyConvert(time3, s_strGMT, s_strTonga, time3.AddHours(13));
             VerifyConvert(time3, s_strTonga, s_strGMT, time3.AddHours(-12));
+
+            // Iran has Utc offset 4:30 during the DST and 3:30 during standard time.
+            var time4 = new DateTime(2018, 4, 20, 7, 0, 0, DateTimeKind.Utc);
+            var time5 = new DateTime(2018, 4, 20, 11, 30, 0, DateTimeKind.Unspecified); // DST time
+            VerifyConvert(time4, s_strIran, time5);
```
Comment (eerhardt) [inline src/System.Runtime/tests/System/TimeZoneInfoTests.cs]: Is there a reason you aren't using `AddHours` like the above code is?  You should be able to say `4.5` and `3.5` below.

**Comment by tarekgh** [comment]
the failure in NETFX x86 Release Build is tracked by the issue #28878

**Comment by tarekgh** [comment]
@eerhardt please let me know if you have any more comments. and thanks for your review.

**Pull request merged**